uber/models.py

```diff
@@ -1109,13 +1109,13 @@ def _staffing_adjustments(self):
             old_staffing = self.orig_value_of('staffing')
             if self.staffing and not old_staffing or self.ribbon == c.VOLUNTEER_RIBBON and old_ribbon != c.VOLUNTEER_RIBBON:
                 self.staffing = True
```

Comment:
By putting the if-statement here, it would only fire if their staffing or ribbon status changed, so we never reached this when all that changed was their badge type.